ibc/modules/src/applications/transfer/acknowledgement.rs

```diff
@@ -13,66 +13,51 @@
 // limitations under the License.
 
 use super::error::Error;
-use crate::{core::ics26_routing::context::Acknowledgement as AckTrait, prelude::*};
+use crate::prelude::*;
 use core::fmt::{Display, Formatter};
 
-use serde::{Deserialize, Deserializer};
+use serde::{Deserialize, Serialize};
 
 /// A string constant included in error acknowledgements.
 /// NOTE: Changing this const is state machine breaking as acknowledgements are written into state
 pub const ACK_ERR_STR: &str = "error handling packet on destination chain: see events for details";
-pub const ACK_SUCCESS_B64: &[u8] = b"AQ==";
-
-#[derive(Clone, Debug)]
-pub enum Acknowledgement {
-	/// Equivalent to b"AQ==" (i.e. `base64::encode(0x01)`)
-	Success(Vec<u8>),
-	/// Error Acknowledgement
-	Error(String),
+pub const ACK_SUCCESS_B64: &str = "AQ==";
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub struct Acknowledgement {
+	#[serde(skip_serializing_if = "Option::is_none")]
+	result: Option<String>,
+	#[serde(skip_serializing_if = "Option::is_none")]
+	error: Option<String>,
 }
 
 impl Acknowledgement {
 	pub fn success() -> Self {
-		Self::Success(ACK_SUCCESS_B64.to_vec())
+		Self { result: Some(ACK_SUCCESS_B64.to_string()), error: None }
 	}
 
 	pub fn from_error(err: Error) -> Self {
-		Self::Error(format!("{}: {}", ACK_ERR_STR, err))
+		Self { result: None, error: Some(err.to_string()) }
```

Comment:
By changing this aren’t we going to have problems with other implementations? An enum is different to a struct. 
Just worried about introducing a new protocol that nobody else other than pallet ibc would understand.

Reply:
They're  using a struct now in ibc-go

Reply:
https://github.com/cosmos/ibc/tree/main/spec/core/ics-004-channel-and-packet-semantics#acknowledgement-envelope
[type Acknowledgement struct {](https://github.com/cosmos/ibc-go/blob/b2fb1192245134607e0e7a294d09d2aa1145f017/modules/core/04-channel/types/channel.pb.go#L402)

Reply:
True. Go has no enums. I wanted to confirm that we weren’t introducing a conflict

Reply:
I can make it an enum, but then serialization will be implemented manually (not through `derive`)

Reply:
Actually, no, I can make it an enum